src/parse/asp/builtins.go

```diff
@@ -678,9 +678,12 @@ func glob(s *scope, args []pyObject) pyObject {
 	if !allowEmpty && len(glob) == 0 {
 		// Strip build file name from exclude list for error message
 		exclude = exclude[:len(exclude)-len(s.state.Config.Parse.BuildFileName)]
-		log.Fatalf("glob(include=%s, exclude=%s) in %s returned no files. If this is intended, set allow_empty=True on the glob.", include, exclude, s.pkg.Filename)
+		if s.state.KeepGoing {
+			log.Warningf("glob(include=%s, exclude=%s) in %s returned no files; continuing due to enabled KeepGoing option.", include, exclude, s.pkg.Filename)
+		} else {
+			log.Fatalf("glob(include=%s, exclude=%s) in %s returned no files. If this is intended, set allow_empty=True on the glob.", include, exclude, s.pkg.Filename)
+		}
```

Comment:
Im not sure but will this not cause a deadlock scenario if there are dependents of the rule that contains the problematic glob?

Reply:
The code path introduced in this change is equivalent to explicitly passing the `allow_empty = True` argument to the glob. Now new deadlocks introduced.